packages/core/src/DdSdkReactNative.tsx

```diff
@@ -204,12 +211,13 @@ export class DdSdkReactNative {
      * @returns a Promise.
      */
     // eslint-disable-next-line @typescript-eslint/ban-types
-    static setUser(user: object): Promise<void> {
+    static async setUser(user: UserInfo): Promise<void> {
         InternalLog.log(
             `Setting user ${JSON.stringify(user)}`,
             SdkVerbosity.DEBUG
         );
-        return DdSdk.setUser(user);
+        await DdSdk.setUser(user);
+        UserInfoSingleton.getInstance().setUserInfo(user);
```

Comment:
Yes it is intended.
We used to `return DdSdk.setUser(user)` which itself returns a `Promise<void>` (promise resolving a value equal to `undefined`).
Now that we turned the function into an `async` one, and used `await`, it also returns a `Promise`, which in this case is a `Promise<void>` as we don't return anything.
This would be equivalent to `return;` or `return undefined` but the more common syntax is to not use a return statement.